File: civitasJAVA/src/civitas/CasillaImpuesto.java
package civitas;

import java.util.ArrayList;

/**
 * @author Jesús Navarro Merino
 * @author José Alberto Gómez García
 */

public class CasillaImpuesto extends Casilla{

    private float importe;
    
    public CasillaImpuesto(String nombre_introducido,float _importe) {
        super(nombre_introducido);
        importe=_importe;
    }
    
    @Override
    void recibeJugador(int actual,ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos); 
            todos.get(actual).pagaImpuesto(importe);
        }
    }
    
    @Override
    public String toString(){
        return super.toString() 
                + "\n\t Importe: " + importe;
    }
    
} //Fin de la clase CasillaImpuesto

File: civitasRUBY/lib/casilla_impuesto.rb
# encoding:utf-8

module Civitas
  class CasillaImpuesto < Casilla
    attr_reader :importe, :nombre
    
    def initialize(_nombre,_importe)
      super(_nombre)
      @importe=_importe
    end
    
    def recibeJugador(actual,todos)
      if(jugadorCorrecto(actual,todos))
        informe(actual,todos)
        (todos[actual]).pagaImpuesto(@importe)
      end
    end
    
    def toString
      super
    end
    
    def self.constructorCasillaImpuesto(_nombre,_importe)
      new(_nombre,_importe)
    end

  end
end
